spec/models/product_importer_spec.rb

```diff
@@ -169,11 +169,28 @@
     end
   end
 
+  describe "when uploading a spreadsheet with some malformed data" do
+    # Use a simple string as CVS.generate will do some escaping
+    let(:csv_data) {
+      csv = "name,producer,category,on_hand,price,units,unit_type,shipping_category\n"
+      csv += "Good Carrots,#{enterprise.name},Vegetables,5,3.20,500,g,#{shipping_category.name}\n"
+      csv += "Malformed \rBrocolli,#{enterprise.name},Vegetables,8,2.50,200,g,#{shipping_category.name}\n"
+    }
+    let(:importer) { import_data csv_data }
+
+    # NOTE an unquoted \n will create a non valid line which will fail entry validation hence why we are only testing with \r
+    it "should raise an unquoted field error if data include unquoted filed with \r character" do
```

Comment:
typo in filed

Reply:
NOTE is not needed, you can just right your comment after #

Reply:
I dont understand "we are only testing with \r"
The test csv contains \n
Why do you say you are only testing with \r?

Reply:
The test data has \n at the end of the lines to create a "valid" csv file. I mentioned we only test with \r as adding a \n in a random place will not trigger the error we are testing, only a \r will. Does that make sense ? Maybe I need to reword my comment.